api_tests/lib/wallets/ethereum.ts

```diff
@@ -42,58 +42,68 @@ export class EthereumWallet implements Wallet {
     }
 
     private async mintErc20(asset: Asset): Promise<void> {
-        let toAddress = this.inner.getAccount();
-
-        const functionIdentifier = "40c10f19";
-        toAddress = toAddress.replace(/^0x/, "").padStart(64, "0");
-
-        const bigNumber = ethers.utils.bigNumberify(asset.quantity);
-        const hexAmount = bigNumber
-            .toHexString()
-            .replace(/^0x/, "")
-            .padStart(64, "0");
-        const data = "0x" + functionIdentifier + toAddress + hexAmount;
-
-        const tx: TransactionRequest = {
-            to: asset.token_contract,
-            gasLimit: "0x100000",
-            value: "0x0",
-            data,
-        };
-        const transactionResponse = await this.parity.sendTransaction(tx);
-        const transactionReceipt = await transactionResponse.wait(1);
-
-        if (!transactionReceipt.status) {
-            throw new Error(
-                `Minting ${asset.quantity} tokens to address ${toAddress} failed`
-            );
-        }
+        const release = await global.parityAccountMutex.acquire();
```

Comment:
Mutexes should be held for as short as possible (in this case, just the `this.parity.sendTransaction` call.

If you extract that in a function, you can actually reuse it between ether and erc20.

Reply:
Good catch.